plaid/src/apis/okta/mod.rs

```diff
@@ -1,20 +1,31 @@
 mod groups;
 mod users;
 
+use jwt_simple::{claims::Claims, prelude::{Duration as JwtDuration, RS256KeyPair, RSAKeyPairLike}};
+use once_cell::sync::Lazy;
 use reqwest::Client;
+use serde::{Deserialize, Serialize};
 
-use serde::Deserialize;
-
-use std::{string::FromUtf8Error, time::Duration};
+use std::{string::FromUtf8Error, sync::Mutex, time::Duration};
 
 use super::default_timeout_seconds;
 
+#[derive(Deserialize)]
+#[serde(untagged)]
+enum Authentication {
+    ApiKey { token: String },
+    OktaApp {
+        client_id: String,
+        private_key: String
```

Comment:
Make this `RS256KeyPair` using a custom serde deserializer. That should simplify the rest of the logic. Then I will finish my final review.

Reply:
Done!